src/interactivity/onecore/ConIoSrvComm.cpp

```diff
@@ -71,12 +71,11 @@ ConIoSrvComm::~ConIoSrvComm()
 
 [[nodiscard]] NTSTATUS ConIoSrvComm::Connect()
 {
-    BOOL Ret = TRUE;
     NTSTATUS Status = STATUS_SUCCESS;
 
     // Port handle and name.
     HANDLE PortHandle;
-    UNICODE_STRING PortName;
+    static UNICODE_STRING PortName = RTL_CONSTANT_STRING(CIS_ALPC_PORT_NAME);

```

Comment:
`static constexpr`? If not then `static const`.

Reply:
This requires a deadly const_cast -- if that is okay, let's do constexpr!

```diff
diff --git a/onecore/windows/core/console/open/src/interactivity/onecore/ConIoSrvComm.cpp b/onecore/windows/core/console/open/src/interactivity/onecore/ConIoSrvComm.cpp
index cdeebed38c365..6c1b4ffc465b0 100644
--- a/onecore/windows/core/console/open/src/interactivity/onecore/ConIoSrvComm.cpp
+++ b/onecore/windows/core/console/open/src/interactivity/onecore/ConIoSrvComm.cpp
@@ -153,7 +145,7 @@ ConIoSrvComm::~ConIoSrvComm()
     // Request to connect to the server.
     ConnectionMessageLength = sizeof(CIS_MSG);
     Status = NtAlpcConnectPort(&PortHandle,
-                               &PortName,
+                               const_cast<PUNICODE_STRING>(&PortName),^M
                                NULL,
                                &PortAttributes,
                                ALPC_MSGFLG_SYNC_REQUEST,
```

Reply:
In that case I'd personally use a regular static... Better safe than sorry, since it's possible it might do reference counting on it, right?